core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/edit-ema-editor.component.ts

```diff
@@ -162,7 +162,13 @@ export class EditEmaEditorComponent implements OnInit, OnDestroy {
             this.dialogIframe.nativeElement.contentWindow.document,
             'ng-event'
         )
-            .pipe(takeUntil(this.destroy$))
+            .pipe(
+                takeUntil(this.destroy$),
+                distinctUntilChanged(
+                    (prev: CustomEvent, current: CustomEvent) =>
+                        prev.detail.name === current.detail.name
+                )
+            )
```

Comment:
This will not work because we do have some cases where the same event (name) can come multiples times and needs to be handled.

For example:

1. Click on the edit button of a contentlet
2. Change the title or any other field in the contentlet
3. Hit Save and Publish (trigger a custom event) **(Don't close the dialog)**
5. Angular catches this event and refreshes the state or whatever needs to be done
6. Now change another field in the contentlet
7. Hit Save and Publish again
8. Because it was the same custom (same name), the subscribe will not emit, and the event will not be handled again.

So even tho this looks like an easy fix in the Angular side, this is a problem on the JSP side.

We need to figure it out why the “select” custom event triggers twice when the user only clicks the “Select” button once.

Reply:
Oh, i see, so now the change is in JSP side (ContentSelector.js)